cmd/main.go

```diff
@@ -163,6 +168,16 @@ func main() {
 		reconcilerCfg.DefaultProfile = "default"
 	}
 
+	// Detect FIPS enabled mode on OpenShift
+	if reconcilerCfg.Platform == config.PlatformOpenShift {
+		contents, err := os.ReadFile(FipsEnableFilePath)
+		fipsEnabled := strings.TrimSuffix(string(contents), "\n")
+		if err == nil && fipsEnabled == "1" {
+			setupLog.Info("FIPS mode is enabled in the system.")
+			istiovalues.FipsEnabled = true
+		}
+	}
+
```

Comment:
It would be great if you moved this to a separate package (e.g. `fips`), since this is too low-level code for the main function.

Reply:
I have updated this and refactored the code. The step above is in a function of istiovalues/fips.go file.